### PR TITLE
Add attribute for media type validation

### PR DESCRIPTION
**Pull request by Shane32**
Requires GraphQL.NET 8.x

Other changes required since extension methods were moved from `IProvideMetadata` to `IMetadataReader` in v8.

~Tests not finished yet~

### CODE REVIEW

File: src/Transports.AspNetCore/Transports.AspNetCore.csproj [hunk 1]
```diff
@@ -8,7 +8,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="GraphQL.MicrosoftDI" Version="$(GraphQLVersion)" />
+    <PackageReference Include="GraphQL.MicrosoftDI" Version="8.0.0-preview-975" />
```
Comment (Shane32) [inline src/Transports.AspNetCore/Transports.AspNetCore.csproj:11]:
```suggestion
    <PackageReference Include="GraphQL.MicrosoftDI" Version="$(GraphQLVersion)" />
```

File: README.md [hunk 2]
```diff
@@ -1001,11 +1001,12 @@ services.AddGraphQL(b => b
...
 ```
 
-Please see the 'Upload' sample for a demonstration of this technique.  Note that
-using the `FormFileGraphType` scalar requires that the uploaded files be sent only
-via the `multipart/form-data` content type as attached files.  If you wish to also
-allow clients to send files as base-64 encoded strings, you can write a custom scalar
-better suited to your needs.
+Please see the 'Upload' sample for a demonstration of this technique, which also
+demonstrates the use of the `MediaTypeAttribute` to restrict the allowable media
+types that will be accepted.  Note that using the `FormFileGraphType` scalar requires
```
Comment (gao-artur) [inline README.md:1006]:
```suggestion
types that will be accepted. Note that using the `FormFileGraphType` scalar requires
```

File: README.md [hunk 3]
```diff
@@ -1001,11 +1001,12 @@ services.AddGraphQL(b => b
...
+that the uploaded files be sent only via the `multipart/form-data` content type as
+attached files.  If you wish to also allow clients to send files as base-64 encoded
```
Comment (gao-artur) [inline README.md:1008]:
```suggestion
attached files. If you also wish to allow clients to send files as base-64 encoded
```

File: src/Transports.AspNetCore/MediaTypeAttribute.cs [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+            if (obj == null)
+                return;
+            if (lists == 0)
+            {
+                if (obj is IFormFile file)
+                    ValidateMediaType(file);
+                else
+                    throw new InvalidOperationException("Expected an IFormFile object.");
+            }
+            else if (obj is IEnumerable enumerable)
```
Comment (gao-artur) [inline src/Transports.AspNetCore/MediaTypeAttribute.cs:73]: Need to exclude the `string` type?

File: docs/migration/migration8.md [hunk 5]
```diff
@@ -1,16 +1,17 @@
...
-None
+- When using `FormFileGraphType` with type-first schemas, you may specify the allowed media
+  types for the file by using the new `[MediaType]` attribute on the argument or input object field.
 
 ## Breaking changes
 
 - The validation rules' signatures have changed slightly due to the underlying changes to the
-  GraphQL.NET library. Please see the GraphQL.NET v8 migration document for more information.
-- The obsolete (v6 and prior) authorization validation rule has been removed. See the v7 migration
+  GraphQL.NET library.  Please see the GraphQL.NET v8 migration document for more information.
```
Comment (gao-artur) [inline docs/migration/migration8.md:11]:
```suggestion
  GraphQL.NET library. Please see the GraphQL.NET v8 migration document for more information.
```

File: docs/migration/migration8.md [hunk 6]
```diff
@@ -1,16 +1,17 @@
...
+- The obsolete (v6 and prior) authorization validation rule has been removed.  See the v7 migration
```
Comment (gao-artur) [inline docs/migration/migration8.md:12]:
```suggestion
- The obsolete (v6 and prior) authorization validation rule has been removed. See the v7 migration
```

File: README.md (hunk 2 above)
Comment (Shane32) [reply README.md:1006]: This repository has "properly" two spaces after every period.

File: README.md (hunk 3 above)
Comment (Shane32) [reply README.md:1008]: This repository has "properly" two spaces after every period consistently throughout all documentation.

Comment (Shane32) [reply README.md:1008]:
```suggestion
attached files.  If you also wish to allow clients to send files as base-64 encoded
```

File: docs/migration/migration8.md (hunk 5 above)
Comment (Shane32) [reply docs/migration/migration8.md:11]: This repository has "properly" two spaces after every period consistently throughout all documentation.

File: docs/migration/migration8.md (hunk 6 above)
Comment (Shane32) [reply docs/migration/migration8.md:12]: This repository has "properly" two spaces after every period consistently throughout all documentation.

File: README.md (hunk 3 above)
Comment (Shane32) [reply README.md]: Merged change to move 'also'

File: README.md (hunk 2 above)
Comment (gao-artur) [reply README.md:1006]: Got it!

File: src/Transports.AspNetCore/MediaTypeAttribute.cs (hunk 4 above)
Comment (Shane32) [reply src/Transports.AspNetCore/MediaTypeAttribute.cs:73]: Sure, we can do that.

Comment (Shane32) [reply src/Transports.AspNetCore/MediaTypeAttribute.cs:73]:
```suggestion
            else if (obj is IEnumerable enumerable && obj is not string)
```

**Review by gao-artur — APPROVED** [review]

**Pull request merged**